### PR TITLE
Document resource identifiers

### LINKED ISSUE #389: Document resource UUIDs
### Issue To Be Solved

All resources are assigned a unique ID, which can be queried using the `let uuid: UInt64` field.

This feature is implemented, but currently not documented.

### Definition of Done

- [ ] Document the feature in the "Composite Types" documentation page

### PR DESCRIPTION
**Pull request by turbolent**
Closes #389

### CODE REVIEW

File: docs/language/composite-types.mdx [hunk 1]
```diff
@@ -1294,6 +1294,40 @@ destroy old2
 destroy resources
 ```
 
+### Resource Identifier
+
+Resources have an impicit unique identifier associated with them,
+all resources have a predeclared public field `let uuid: UInt64`.
```
Comment (jordanschalm) [inline docs/language/composite-types.mdx:1300]:
```suggestion
implemented by a predeclared public field `let uuid: UInt64` on each resource.
```

**Review by jordanschalm — APPROVED** [review]
LGTM, added one phrasing suggestion

File: docs/language/composite-types.mdx [hunk 2]
```diff
@@ -1294,6 +1294,40 @@ destroy old2
 destroy resources
 ```
 
+### Resource Identifier
+
+Resources have an impicit unique identifier associated with them,
```
Comment (turbolent) [inline docs/language/composite-types.mdx:1299]:
```suggestion
Resources have an implicit unique identifier associated with them,
```

**Comment by turbolent** [comment]
Thanks for the review @jordanschalm!

**Comment by turbolent** [comment]
@joshuahannan could you please have a look?

**Review by joshuahannan — APPROVED** [review]
Can you explain how identifiers are assigned and give an example?

**Pull request merged**